symbulate/distributions.py

```diff
@@ -618,16 +666,33 @@ def __init__(self,
                  sd1=1.0, sd2=1.0, corr=0.0,
                  var1=None, var2=None, cov=None):
 
-        if corr is not None and not (-1 <= corr < 1):
+        if corr is not None and not (-1 <= corr <= 1):
             raise Exception("Correlation must be "
                             "between -1 and 1.")
 
-        self.mean = [mean1, mean2]
+        if (isinstance(mean1, float)) and (isinstance(mean2, float)):    
+            self.mean = [mean1, mean2]
+        elif isinstance(mean1, float):
+            raise Exception("mean2 argument must be a float")
+        else:
+            raise Exception("mean1 argument must be a float")
+
+        if (sd1 < 0):
+            raise Exception("sd1 cannot be less than 0")
+        if (sd2 < 0):
+            raise Exception("sd2 cannot be less than 0")
+
 
         if var1 is None:
             var1 = sd1 ** 2
+        else:
+            if var1 < 0:
+                raise Exception("var1 cannot be negative")
         if var2 is None:
             var2 = sd2 ** 2
+        else:
+            if var2 < 0:
+                raise Exception("var2 cannot be negative")
         if cov is None:
             if var1 * var2 < 0:
```

Comment:
This is a weird check. In particular, you bypass the exception if both var1 and var2 are negative, which seems wrong. Instead, you should raise the exception if _either_ var1 or var2 is less than 0.